tck/src/main/java/org/eclipse/microprofile/fault/tolerance/tck/telemetryMetrics/BulkheadTelemetryTest.java

```diff
@@ -206,24 +203,24 @@ public void bulkheadMetricHistogramTest() throws InterruptedException, Execution
         assertThat("histogram count", executionTimesCount, is(2L)); // Rejected executions
                                                                     // not recorded in
 
-        Collection<HistogramPointData> executionTimesPoints = m.getBulkheadRunningDuration().getHistogramPoints();
-        double time = executionTimesPoints.stream()
-                .mapToDouble(points -> points.getSum())
-                .sum();
+        Optional<HistogramPointData> executionTimesPoint = m.getBulkheadRunningDuration().getHistogramPoint();
+        double time = executionTimesPoint
+                .map(HistogramPointData::getSum)
+                .orElse(0.0);
 
-        long count = executionTimesPoints.stream()
-                .mapToLong(points -> points.getCount())
-                .sum();
+        long count = executionTimesPoint
+                .map(HistogramPointData::getCount)
+                .orElse(0L);
 
-        assertThat("mean", Math.round(time / count), approxMillis(1000)); // histogram
+        assertThat("mean", time / count, approxMillisFromSeconds(1000)); // histogram
 
         // Now let's put some quick results through the bulkhead
         bulkheadBean.waitForHistogram(CompletableFuture.completedFuture(null));
         bulkheadBean.waitForHistogram(CompletableFuture.completedFuture(null));
 
         // Should have 4 results, ~0ms * 2 and ~1000ms * 2
-        executionTimesCount = m.getBulkheadRunningDuration().getHistogramCount().get();
-        assertThat("histogram count", executionTimesCount, is(4L));
+        m.getBulkheadRunningDuration().assertBucketCounts(1000, 1000, 0, 0);
```

Comment:
I wonder if we should go back to the old metric test and also make that use buckets? Not urgent of course.

Reply:
That would be a good idea in the future, but at the moment the TCK compiles against MP Metrics 4.0 which doesn't have explicit buckets, it just has percentiles (which we can't reliably test, see message on 28d305).

If we set a minimum version of MP Metrics in a future release, we could add an assertion like this to `BulkheadMetricTest`.